git-multimail/git_multimail.py

```diff
@@ -3756,11 +3759,13 @@ def run_as_post_receive_hook(environment, mailer):
         changes.append(
             ReferenceChange.create(environment, oldrev, newrev, refname)
             )
-    if changes:
-        push = Push(environment, changes)
+    if not changes:
+        return
+    push = Push(environment, changes)
+    try:
         push.send_emails(mailer, body_filter=environment.filter_body)
-    if hasattr(mailer, '__del__'):
-        mailer.__del__()
+    finally:
```

Comment:
This try/finally is a good change, but ideally it would go to another commit. No strong objection if you prefer keeping it in the same commit though.

Reply:
I can split it.